app/src/main/java/com/poupa/vinylmusicplayer/ui/fragments/player/card/CardPlayerFragment.java

```diff
@@ -529,10 +530,10 @@ public void setUpPanelAndAlbumCoverHeight() {
         }
 
         @Override
-        public void updateCurrentSong(Song song) {
-            currentSong = song;
-            currentSongViewHolder.title.setText(song.title);
-            currentSongViewHolder.text.setText(MusicUtil.getSongInfoString(song));
+        public void updateCurrentSong(IndexedSong song) {
+            currentSong = song.song;
```

Comment:
Same comment as above: We dont need the concept of IndexedSong here, and by having it as a subclass of Song, there is no need to change the code nor writing `song.song.title` which is kind of awkward :-)

Reply:
we ned IndexedSong there for the isPlaying function, if we don't have that, and there is two instance of a song in the playing queue, isPlaying will be wrongly true for one of them